lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,186 @@
+  # UtilsParser class
+  class UtilsParser
+    # Converts string and float input to int. Strips all non-numeric
+    # characters from strings.
+    #
+    # @param value [String, Float] Input value
+    #
+    # @return [Integer, nil] Integer if successful conversion, otherwise nil
+    def self.convert_to_int(value)
+      if value.is_a?(String)
+        str_val = value.gsub(/[^0-9\-.]/, '')
+        begin
+          Integer(str_val)
+        rescue ArgumentError, TypeError
+          begin
+            Integer(Float(str_val))
+          rescue ArgumentError, TypeError
+            nil
+          end
+        end
+      elsif value.is_a?(Numeric)
+        value.to_i
+      end
+    end
+  end
+
+  # SysInfoParser class
+  class SysInfoParser
+    DELIM = ':' # k/v delimiter
+
+    # Main text parsing function
+    #
+    # @param data [String] Text data to parse
+    #
+    # @return [Array<Hash>] List of dictionaries. Processed structured data.
+    def parse(data)
+      lines = data.lines.map(&:chomp).reject(&:empty?)
+
+      # find the character position of the value in the k/v pair of the first line
+      # all subsequent lines of data use the same character position
+      start_value_pos = get_value_pos(lines[0], DELIM)
+
+      raw_data = build_raw_data(lines, start_value_pos)
+      build_data(raw_data)
+    end
+
+    private
+
+    # Build k/v hash without parsing complex structures
+    #
+    # @param lines [String] Input string without empty
+    # @param start_value_pos [Integer] Start position of value string
+    def build_raw_data(lines, start_value_pos)
+      raw_data = {}
+
+      last_key = nil
+      lines.each do |line|
+        key = line[0...start_value_pos]
+        value = line[start_value_pos..]
+
+        # possible multiline data
+        # the value data doesn't start where it should
+        # so this is multiline data
+        if last_key && !(key.include? DELIM)
+          raw_data[last_key] += "\n#{line}"
+          next
+        end
+
+        raw_data[key] = value
+        last_key = key
+      end
+
+      raw_data
+    end
+
+    # Build k/v hash without parsing comples structures
+    #
+    # @param raw_data [Hash] k/v hash with unparsed complex structures
+    def build_data(raw_data)
+      raw_output = {}
+      raw_data.each do |k, v|
+        # clean up keys
+        k = transform_key(k)
+        # since we split on start_value_pos, the delimiter
+        # is still in the key field. Remove it.
+        k = k.chomp(DELIM)
+        case k
+        # network_cards currently are not parsed
+        when 'hotfixs', 'processors'
+          raw_output[k] = parse_hotfixs_or_processors(v)
+        when 'hyperv_requirements'
+          raw_output[k] = parse_hyperv_requirements(v)
+        when 'total_physical_memory', 'available_physical_memory', 'virtual_memory_max_size',
+             'virtual_memory_available', 'virtual_memory_in_use'
+          raw_output["#{k}_mb"] = UtilsParser.convert_to_int(v)
```

Comment:
The methods called here are so complex. I suggest to remove them and add later if necessary.

Reply:
Currently we need to parse a processors at least.

Reply:
That makes sense. Remove the others then.